### PR TITLE
Remove opcode limits

### PR DESCRIPTION
**Pull request by jpcima**
A work in progress

### CODE REVIEW

**Comment by jpcima** [comment]
Most limits are lifted, except key and fx for now.
The former has a bit more complicated logic involved and I've not checked each single case.

Some opcodes remain oob-defaulted, it's the cases of enums and identifiers (for example cc numbers).

Ok for review

**Comment by paulfd** [comment]
Wouldn't it make sense to just have permissive bounds as default?

File: src/sfizz/Range.h [hunk 1]
```diff
@@ -32,6 +32,12 @@ class Range {
     {
 
     }
+
+    constexpr Range(const Range<Type, !Checked>& other)
```
Comment (paulfd) [inline src/sfizz/Range.h:36]: Why the `!Checked` here?

File: src/sfizz/Range.h [hunk 2]
```diff
@@ -32,6 +32,12 @@ class Range {
...
+        : Range(other.getStart(), other.getEnd())
+    {
+    }
+
     constexpr Type getStart() const noexcept { return _start; }
```
Comment (paulfd) [inline src/sfizz/Range.h:41]: In the rest of the file all the `if (Checked) ...` could use the `IF_CONSTEXPR` macro for C++17, I think we have it.

**Review by paulfd — APPROVED** [review]

**Comment by jpcima** [comment]
>Wouldn't it make sense to just have permissive bounds as default?

This would be the best idea starting from scratch, but I thought this easier for transition purposes.

File: src/sfizz/Range.h (hunk 1 above)
Comment (jpcima) [reply src/sfizz/Range.h:36]: To allow `UncheckedRange` to be converted to `Range` and vice-versa

**Pull request merged**